Packer 1.2.5 fails for AWS with 'ResourceNotReady: exceeded wait attempts'

I am shameless copying this issue but updating slightly: https://github.com/hashicorp/packer/issues/5792

It may be related to https://github.com/hashicorp/packer/issues/6526

I have a CentOS Packer build on AWS which works with version 1.1.2 but fails with version 1.2.5

The template used to build the image:

```
{
  "variables": {
    "packer_script_path": "",
    "force_deregister": "",
    "image_name": "",
    "aws_secret_key_id": "{{env `AWS_SECRET_KEY_ID`}}",
    "aws_secret_key": "{{env `AWS_SECRET_ACCESS_KEY`}}",
    "source_ami": "ami-476e4c22",
    "ami_regions": "us-west-2",
    "ssh_username": "{{env `DEFAULT_PLATFORM_USER`}}"
  },
  "builders": [
    {
      "name": "aws",
      "type": "amazon-ebs",
      "region": "us-east-2",
      "source_ami": "{{user `source_ami`}}",
      "instance_type": "t2.medium",
      "ssh_username": "{{user `ssh_username`}}",
      "ssh_pty": "true",
      "run_tags": {
        "Source": "Packer",
      },
      "tags": {
        "Source": "Packer",
      },
      "ami_name": "{{user `image_name`}}",
      "ami_description": "{{user `image_name`}}",
      "force_deregister": "{{user `force_deregister`}}",
      "ami_regions": "{{user `ami_regions`}}",
      "shutdown_behavior": "terminate"
    }
  ],
  "provisioners": [
    {
      "type": "shell",
      "execute_command": "{{ .Vars }} sudo -E -S sh '{{ .Path }}'",
      "scripts": [
        "{{user `packer_script_path`}}/install-packages.sh"
      ]
    },
    {
      "type": "shell",
      "execute_command": "{{ .Vars }} sudo -E -S sh '{{ .Path }}'",
      "scripts": [
        "{{user `packer_script_path`}}/configure-os.sh"
      ]
    },
    {
      "type": "shell",
      "execute_command": "{{ .Vars }} sudo -E -S sh '{{ .Path }}'",
      "scripts": [
        "{{user `packer_script_path`}}/set-kernel.sh"
      ]
    }
  ],
  "post-processors": [
    {
      "type": "manifest",
      "output": "image-metadata/aws-{{user `file_name`}}.json",
      "strip_path": true
    }
  ]
}
```

As you can see, I am not doing anything complicated.

When the build fails using version 1.2.5, it times out with the following error. This happens 75% of the time:

```
==> aws: 1 error(s) occurred:
==> aws: 
==> aws: * Error waiting for AMI (ami-192c0861) in region (us-west-2): ResourceNotReady: exceeded wait attempts
```

* Packer version from packer version
  1.2.5

* Host platform
  Alpine Linux 3.8.0

I am using this image from dockerhub, https://hub.docker.com/r/hashicorp/packer/, with a tag of `light`.

If I run the above template using v 1.1.2, it works fine.

Regard,
Kris

It doesn't appear that they do. Here's my log, truncated from the time the VM is shut down:
```
2018/08/30 12:42:31 ui: ==> virtualbox-iso: Gracefully halting virtual machine...
2018/08/30 12:42:31 packer: 2018/08/30 12:42:31 Executing shutdown command: sudo -S shutdown -P now
2018/08/30 12:42:31 packer: 2018/08/30 12:42:31 [DEBUG] Opening new ssh session
2018/08/30 12:42:31 packer: 2018/08/30 12:42:31 [DEBUG] starting remote command: sudo -S shutdown -P now
2018/08/30 12:42:37 packer: 2018/08/30 12:42:37 [ERROR] Remote command exited without exit status or exit signal.
2018/08/30 12:42:37 packer: 2018/08/30 12:42:37 Waiting max 5m0s for shutdown to complete
2018/08/30 12:42:38 packer: 2018/08/30 12:42:38 VM shut down.
2018/08/30 12:42:38 packer: 2018/08/30 12:42:38 Executing VBoxManage: []string{"storageattach", "packer_centos_7_5b884589-9135-e87b-fc60-ce7d05873bb1", "--storagectl", "IDE Controller", "--port", "0", "--device", "1", "--medium", "none"}
2018/08/30 12:42:38 packer: 2018/08/30 12:42:38 stdout:
2018/08/30 12:42:38 packer: 2018/08/30 12:42:38 stderr:
2018/08/30 12:42:38 ui: ==> virtualbox-iso: Executing custom VBoxManage commands...
2018/08/30 12:42:38 ui:     virtualbox-iso: Executing: modifyhd --compact ./output/packer_centos_7_5b884589-9135-e87b-fc60-ce7d05873bb1.vdi
2018/08/30 12:42:38 packer: 2018/08/30 12:42:38 Executing VBoxManage: []string{"modifyhd", "--compact", "./output/packer_centos_7_5b884589-9135-e87b-fc60-ce7d05873bb1.vdi"}
2018/08/30 12:42:38 packer: 2018/08/30 12:42:38 stdout:
2018/08/30 12:42:38 packer: 2018/08/30 12:42:38 stderr: 0%...10%...20%...30%...40%...50%...60%...70%...80%...90%...100%
2018/08/30 12:42:38 packer: 2018/08/30 12:42:38 1 second timeout to ensure VM is really shutdown
2018/08/30 12:42:39 ui: ==> virtualbox-iso: Preparing to export machine...
2018/08/30 12:42:39 ui:     virtualbox-iso: Deleting forwarded port mapping for the communicator (SSH, WinRM, etc) (host port 4275)
2018/08/30 12:42:39 packer: 2018/08/30 12:42:39 Executing VBoxManage: []string{"modifyvm", "packer_centos_7_5b884589-9135-e87b-fc60-ce7d05873bb1", "--natpf1", "delete", "packercomm"}
2018/08/30 12:42:39 packer: 2018/08/30 12:42:39 stdout:
2018/08/30 12:42:39 packer: 2018/08/30 12:42:39 stderr:
2018/08/30 12:42:39 ui: ==> virtualbox-iso: Exporting virtual machine...
2018/08/30 12:42:39 ui:     virtualbox-iso: Executing: export packer_centos_7_5b884589-9135-e87b-fc60-ce7d05873bb1 --output output/packer_centos_7_5b884589-9135-e87b-fc60-ce7d05873bb1.ova
2018/08/30 12:42:39 packer: 2018/08/30 12:42:39 Executing VBoxManage: []string{"export", "packer_centos_7_5b884589-9135-e87b-fc60-ce7d05873bb1", "--output", "output/packer_centos_7_5b884589-9135-e87b-fc60-ce7d05873bb1.ova"}
2018/08/30 12:43:31 packer: 2018/08/30 12:43:31 stdout: Successfully exported 1 machine(s).
2018/08/30 12:43:31 packer: 2018/08/30 12:43:31 stderr: 0%...10%...20%...30%...40%...50%...60%...70%...80%...90%...100%
2018/08/30 12:43:32 packer: 2018/08/30 12:43:32 Executing VBoxManage: []string{"storageattach", "packer_centos_7_5b884589-9135-e87b-fc60-ce7d05873bb1", "--storagectl", "IDE Controller", "--port", "0", "--device", "1", "--medium", "none"}
2018/08/30 12:43:32 packer: 2018/08/30 12:43:32 stdout:
2018/08/30 12:43:32 packer: 2018/08/30 12:43:32 stderr: VBoxManage: error: No storage device attached to device slot 1 on port 0 of controller 'IDE Controller'
2018/08/30 12:43:32 packer: VBoxManage: error: Details: code VBOX_E_OBJECT_NOT_FOUND (0x80bb0001), component SessionMachine, interface IMachine, callee nsISupports
2018/08/30 12:43:32 packer: VBoxManage: error: Context: "DetachDevice(Bstr(pszCtl).raw(), port, device)" at line 403 of file VBoxManageStorageController.cpp
2018/08/30 12:43:32 ui: ==> virtualbox-iso: Deregistering and deleting VM...
2018/08/30 12:43:32 packer: 2018/08/30 12:43:32 Executing VBoxManage: []string{"unregistervm", "packer_centos_7_5b884589-9135-e87b-fc60-ce7d05873bb1", "--delete"}
2018/08/30 12:43:32 packer: 2018/08/30 12:43:32 stdout:
2018/08/30 12:43:32 packer: 2018/08/30 12:43:32 stderr: 0%...10%...20%...30%...40%...50%...60%...70%...80%...90%...100%
2018/08/30 12:43:32 ui: ==> virtualbox-iso: Running post-processor: manifest
2018/08/30 12:43:32 Flagging to keep original artifact from post-processor 'manifest'
2018/08/30 12:43:32 ui: ==> virtualbox-iso: Running post-processor: amazon-import
2018/08/30 12:43:32 packer: 2018/08/30 12:43:32 Error getting region from metadata service, probably because we're not running on AWS. RequestError: send request failed
2018/08/30 12:43:32 packer: caused by: Get http://169.254.169.254/latest/meta-data/placement/availability-zone: dial tcp 169.254.169.254:80: connect: host is down
2018/08/30 12:43:32 packer: 2018/08/30 12:43:32 Found region us-west-2
2018/08/30 12:43:32 packer: 2018/08/30 12:43:32 [INFO] AWS Auth provider used: "SharedConfigCredentials: /Users/molenik/.aws/credentials"
2018/08/30 12:43:32 packer: 2018/08/30 12:43:32 Rendered s3_key_name as tmp/packer/centos-7.5.1804-201808300729.ova
2018/08/30 12:43:32 packer: 2018/08/30 12:43:32 Looking for OVA in artifact
2018/08/30 12:43:32 packer: 2018/08/30 12:43:32 Opening file output/packer_centos_7_5b884589-9135-e87b-fc60-ce7d05873bb1.ova to upload
2018/08/30 12:43:32 ui:     virtualbox-iso (amazon-import): Uploading output/packer_centos_7_5b884589-9135-e87b-fc60-ce7d05873bb1.ova to s3://molenik-ami/tmp/packer/centos-7.5.1804-201808300729.ova
2018/08/30 12:46:27 ui:     virtualbox-iso (amazon-import): Completed upload of output/packer_centos_7_5b884589-9135-e87b-fc60-ce7d05873bb1.ova to s3://molenik-ami/tmp/packer/centos-7.5.1804-201808300729.ova
2018/08/30 12:46:27 packer: 2018/08/30 12:46:27 Calling EC2 to import from s3://molenik-ami/tmp/packer/centos-7.5.1804-201808300729.ova
2018/08/30 12:46:28 ui:     virtualbox-iso (amazon-import): Started import of s3://molenik-ami/tmp/packer/centos-7.5.1804-201808300729.ova, task id import-ami-0e94e4ddc9326f762
2018/08/30 12:46:28 ui:     virtualbox-iso (amazon-import): Waiting for task import-ami-0e94e4ddc9326f762 to complete (may take a while)
2018/08/30 13:11:45 ui: ==> virtualbox-iso: Running post-processor: shell-local
2018/08/30 13:11:45 packer: 2018/08/30 13:11:45 [INFO] (shell-local): Prepending inline script with #!/bin/sh -e
2018/08/30 13:11:45 ui: ==> virtualbox-iso (shell-local): Running local shell script: /var/folders/yd/w7q586lx71s6fnmyylvlkf2ssqhywn/T/packer-shell026291479
2018/08/30 13:11:45 packer: 2018/08/30 13:11:45 [INFO] (shell-local): starting local command: /bin/sh -c PACKER_BUILDER_TYPE='virtualbox-iso' PACKER_BUILD_NAME='virtualbox-iso'  /var/folders/yd/w7q586lx71s6fnmyylvlkf2ssqhywn/T/packer-shell026291479
2018/08/30 13:11:45 packer: 2018/08/30 13:11:45 [INFO] (shell-local communicator): Executing local shell command [/bin/sh -c PACKER_BUILDER_TYPE='virtualbox-iso' PACKER_BUILD_NAME='virtualbox-iso'  /var/folders/yd/w7q586lx71s6fnmyylvlkf2ssqhywn/T/packer-shell026291479]
2018/08/30 13:11:47 ui error: Build 'virtualbox-iso' errored: 1 error(s) occurred:

* Post-processor failed: Import task import-ami-0e94e4ddc9326f762 failed with error: ResourceNotReady: exceeded wait attempts
2018/08/30 13:11:47 Builds completed. Waiting on interrupt barrier...
2018/08/30 13:11:47 machine readable: error-count []string{"1"}
2018/08/30 13:11:47 ui error: 
==> Some builds didn't complete successfully and had errors:
2018/08/30 13:11:47 machine readable: virtualbox-iso,error []string{"1 error(s) occurred:\n\n* Post-processor failed: Import task import-ami-0e94e4ddc9326f762 failed with error: ResourceNotReady: exceeded wait attempts"}
2018/08/30 13:11:47 ui error: --> virtualbox-iso: 1 error(s) occurred:

* Post-processor failed: Import task import-ami-0e94e4ddc9326f762 failed with error: ResourceNotReady: exceeded wait attempts
2018/08/30 13:11:47 ui: 
==> Builds finished but no artifacts were created.
2018/08/30 13:11:47 waiting for all plugin processes to complete...
2018/08/30 13:11:47 /usr/local/bin/packer: plugin process exited
2018/08/30 13:11:47 /usr/local/bin/packer: plugin process exited
2018/08/30 13:11:47 /usr/local/bin/packer: plugin process exited
2018/08/30 13:11:47 /usr/local/bin/packer: plugin process exited
2018/08/30 13:11:47 /usr/local/bin/packer: plugin process exited
2018/08/30 13:11:47 /usr/local/bin/packer: plugin process exited
2018/08/30 13:11:47 /usr/local/bin/packer: plugin process exited
2018/08/30 13:11:47 /usr/local/bin/packer: plugin process exited
2018/08/30 13:11:47 /usr/local/bin/packer: plugin process exited
2018/08/30 13:11:47 [ERR] Error decoding response stream 7: EOF
2018/08/30 13:11:47 /usr/local/bin/packer: plugin process exited
```

I also uploaded the first half of the log up till VM start: https://paste.ubuntu.com/p/7dtH5YGDdD

If I run `env` inside a shell-local provisioner, it shows the AWS variables and the rest of my environment being correctly passed to the process:

```
...
    virtualbox-iso: JABBA_HOME=/Users/molenik/.jabba
    virtualbox-iso: PACKER_RUN_UUID=98c3446c-d7cc-7eaf-79d7-4104ab3dbdc6
    virtualbox-iso: AWS_POLL_DELAY_SECONDS=30
    virtualbox-iso: GOPATH=/Users/molenik/go
...
```

I've tried every combination of vars according to the code but none of them work. The run times out in the same spot every time.

--- a/builder/amazon/common/state.go
+++ b/builder/amazon/common/state.go
@@ -169,6 +169,8 @@ func WaitForVolumeToBeAttached(c *ec2.EC2, ctx aws.Context, input *ec2.DescribeV
 			return req, nil
 		},
 	}
+	w.ApplyOptions(opts...)
+
 	return w.WaitWithContext(ctx)
 }
 
@@ -198,6 +200,8 @@ func WaitForVolumeToBeDetached(c *ec2.EC2, ctx aws.Context, input *ec2.DescribeV
 			return req, nil
 		},
 	}
+	w.ApplyOptions(opts...)
+
 	return w.WaitWithContext(ctx)
 }
 
@@ -227,6 +231,8 @@ func WaitForImageToBeImported(c *ec2.EC2, ctx aws.Context, input *ec2.DescribeIm
 			return req, nil
 		},
 	}
+	w.ApplyOptions(opts...)
+
 	return w.WaitWithContext(ctx)
 }
 
@@ -281,11 +287,11 @@ func getOverride(varInfo envInfo) envInfo {
 	return varInfo
 }
 func getEnvOverrides() overridableWaitVars {
-	// Load env vars from environment, and use them to override defaults
+	// Load env vars from environment.
 	envValues := overridableWaitVars{
-		envInfo{"AWS_POLL_DELAY_SECONDS", 2, false},
+		envInfo{"AWS_POLL_DELAY_SECONDS", 0, false},
 		envInfo{"AWS_MAX_ATTEMPTS", 0, false},
-		envInfo{"AWS_TIMEOUT_SECONDS", 300, false},
+		envInfo{"AWS_TIMEOUT_SECONDS", 0, false},
 	}
 
 	envValues.awsMaxAttempts = getOverride(envValues.awsMaxAttempts)

